src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -71,6 +71,57 @@ public List<TElement> ToList()
             return list;
         }
 
+        public HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer)
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(_source);
```

Comment:
Agree with @JonHanna. This is also the most flexible option since it allows us to change it back to being ordered later.